### PR TITLE
Fix cuda native loss_ctc for varying input length

### LINKED ISSUE #14401: ctc loss get nan after some  epochs in pytorch  1.0.0.dev20181115
## 🐛 Bug

when I train a cnn-rnn-ctc text recognize model, I meet nan loss after some ~~iters~~ epochs, but it's ok at pytorch 0.4 with warpctc

download the code from https://github.com/WenmuZhou/crnn.pytorch
change the ctc loss from warpctc to nn.CTCloss()
run
PyTorch Version (e.g., 1.0): 1.0.0.dev20181115
OS (e.g., Linux): ubuntu 16.04
How you installed PyTorch (conda, pip, source): pip3
Build command you used (if compiling from source):
Python version: 3.5.2
CUDA/cuDNN version: 8.0/6.0
GPU models and configuration: 1080ti

### PR DESCRIPTION
**Pull request by t-vi**
Thank you, @freesouls, for the reproducing example!

This is strictly fixing the bug in gradients for varying length inputs discussed in the middle-to-bottom of the bug report. I'll have a feature patch regarding inf losses -> NaN grads separately.

Fixes: #14401

### CODE REVIEW

**Comment by freesouls** [comment]
glad to hear that, hoping for the new release

**Review by soumith — APPROVED** [review]
